src/averageAges.js

```diff
@@ -55,7 +68,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const moms = people
+    .filter(mother => people.some(child => child.mother === mother.name));
+
+  const children = onlyWithSon
+    ? people
+      .filter(person => moms.some(mother => mother.name === person.mother)
+      && person.sex === 'm')
+    : people
+      .filter(person => moms.some(mother => mother.name === person.mother));
```

Comment:
Move to the separate function 
```suggestion
    ? people.filter(person => hasMother(people, person.mother) && person.sex === 'm')
    : people.filter(person => hasMother(people, person.mother));
```